Source/DafnyCore/Dafny.atg

```diff
@@ -3202,6 +3203,52 @@ Forall = "forall".
 Exists = "exists".
 QSep = "::".
 
+Expression<out Expression e, bool allowLemma, bool allowLambda, bool allowBitwiseOps = true>
+=
+  NotDecreasesToExpression<out e, allowLemma, allowLambda, allowBitwiseOps>
+  .
+
+MaybeDecreasesToExpression<out Expression e, IToken lp>
+= (. List<ActualBinding> args = new();
+     List<Expression> newExprs = new();
+     Expression tmp = null;
+     IToken x = null;
+     bool allowNoChange = false;
+  .)
+  TupleArgs<args>
+  [ ( "decreases"                                                               (. allowNoChange = false; .)
+    | "nonincreases"                                                            (. allowNoChange = true; .)
+    ) (. x = t; .)
+    ident
+    (. if (t.val != "to") {
+         SemErr(ErrorId.p_decreases_without_to, t, "expected 'to'");
+       }
+    .)
+    NotDecreasesToExpression<out tmp, false, false>                             (. newExprs.Add(tmp); .)
```

Comment:
`NotDecreasesToExpression` reads a bit as if this a particular subtype of expression. I would use the term `ExpressionWithoutDecreases`.